cli/ballerina-cli/src/main/java/io/ballerina/cli/diagnostics/AnnotateDiagnostics.java

```diff
@@ -0,0 +1,286 @@
+        currentPackage.moduleIds().forEach(moduleId -> {
+            Module module = currentPackage.module(moduleId);
+            module.documentIds().forEach(documentId -> processDocuments(module, documentId, documentMap));
+            module.testDocumentIds().forEach(documentId -> processDocuments(module, documentId, documentMap));
+        });
+
+        return documentMap;
+    }
+
+    private static String getDocumentPath(ModuleName moduleName, String documentName) {
+        String documentNameFixed = isWindows() ? documentName.replace("/", "\\") : documentName;
+        if (moduleName.isDefaultModuleName()) {
+            return documentNameFixed;
+        }
+        return Paths.get("modules", moduleName.moduleNamePart(), documentNameFixed).toString();
+    }
+
+    private static String diagnosticToString(Diagnostic diagnostic, boolean colorEnabled) {
+        DiagnosticInfo diagnosticInfo = diagnostic.diagnosticInfo();
+        DiagnosticSeverity severity = diagnosticInfo.severity();
+        String severityString = severity.toString();
+        String color = SEVERITY_COLORS.get(severity);
+        String message = diagnostic.toString().substring(severityString.length());
+        String code = diagnosticInfo.code();
+        boolean isMultiline = diagnostic.message().contains(NEW_LINE);
+        boolean isCodeNotNull = code != null;
+        String formatString = getColoredString("%s", color, colorEnabled) + "%s" +
+                (isCodeNotNull ? (isMultiline ? NEW_LINE + "(%s)" : " (%s)") : "");
+
+        return String.format(formatString, severityString, message, isCodeNotNull ? code : "");
+    }
+
+    private static DiagnosticAnnotation getDiagnosticAnnotation(Document document, Location location,
+                                                                DiagnosticSeverity severity, int terminalWidth,
+                                                                boolean colorEnabled) {
+        TextDocument textDocument = document.textDocument();
+        LocationDetails locationDetails = getLocationDetails(location);
+        boolean isMultiline = locationDetails.startLine != locationDetails.endLine;
+        int length = isMultiline ? textDocument.line(locationDetails.startLine).length() - locationDetails.startOffset :
+                locationDetails.endOffset - locationDetails.startOffset;
+
+        return new DiagnosticAnnotation(
+                getLines(textDocument, locationDetails.startLine, locationDetails.endLine),
+                locationDetails.startOffset,
+                length == 0 ? 1 : length,
+                isMultiline,
+                locationDetails.endOffset,
+                locationDetails.startLine + 1,
+                severity,
+                DiagnosticAnnotation.DiagnosticAnnotationType.REGULAR,
+                terminalWidth, colorEnabled);
+    }
+
+    private static DiagnosticAnnotation getSyntaxDiagnosticAnnotation(Document document,
```

Comment:
Do we have any particular reason for making this `static`? Given this is called only from instance methods, if we make this an instance method as well we can avoid explicitly passing instance fields as parameters here.